### ISSUE TITLE
\Request and  relative 301 location

### ISSUE THREAD

**Issue by lindesbs** [contao/core#7799]
Requesting entries from a sitemap http://www.heise.de/sitemap.xml
Those entries are redirected to a relative URL, which is wrong handled by the Request::send function

Array
(
    [scheme] => http
    [host] => www.heise.de
    [path] => /meldung/Hickhack-um-Virtual-Reality-Apps-fuer-Samsungs-Gear-VR-2631443.html
)

Redirect goes to :
Array
(
    [path] => /newsticker/meldung/Hickhack-um-Virtual-Reality-Apps-fuer-Samsungs-Gear-VR-2631443.html
)

that means, the redirect here (https://github.com/contao/core/blob/master/system/modules/core/library/Contao/Request.php#L443) has to check, if scheme, host, port exists and rebuild the URL

**Comment by leofeyer** [comment]
Is this issue still relevant?

**Comment by lindesbs** [comment]
Not for my project, but maybe for the future ? It´s your choice ;-)

**Comment by leofeyer** [comment]
Ok, can you explain the problem to me again please? :)

**Comment by lindesbs** [comment]
example code :
```
$request = new \Request();
$request->send($url);
echo $request->response;
```
sitemap from heise.de, one entry, for example :
```
$url = "http://www.heise.de/meldung/Kunstprojekt-Random-Darknet-Shopper-geht-wieder-auf-Einkaufstour-3024099.html"
```

response is a 301. 

adding redirect option into request :

```
$request = new \Request();
$request->redirect=true;
$request->rlimit=8;
$request->send($url);
echo $request->response;
```
results in 301 too.

This happens cause this line : https://github.com/contao/core/blob/master/system/modules/core/library/Contao/Request.php#L433

The **Location** in the ResponseHeader has only a relative URL : */newsticker/meldung/Kunstprojekt-Random-Darknet-Shopper-geht-wieder-auf-Einkaufstour-3024099.html* 

Now the $uri (https://github.com/contao/core/blob/master/system/modules/core/library/Contao/Request.php#L271) is in the second step only :

```
Array ( [path] => /newsticker/meldung/Kunstprojekt-Random-Darknet-Shopper-geht-wieder-auf-Einkaufstour-3024099.html ) 
```

and without any scheme, the response and everything else stay with old values (https://github.com/contao/core/blob/master/system/modules/core/library/Contao/Request.php#L287)

--> http error 301


I think, it will be sufficient, in generating an asolute URL based on the request header and the response location, if the response location is an relative URL.

**Comment by leofeyer** [comment]
I definitely want to fix this! :)

**Comment by leofeyer** [comment]
Fixed in 6d12f6b146f8a548c0a3c25e5f303f3c35c7ed39.
